Remotion/ObjectBinding/Web/UI/Controls/BocListImplementation/Rendering/BocCellAttributeRenderingContext.cs

```diff
@@ -0,0 +1,111 @@
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+
+using System.Web;
+using System.Web.UI;
+using Remotion.ObjectBinding.Web.Services;
+using Remotion.Utilities;
+using Remotion.Web;
+
+namespace Remotion.ObjectBinding.Web.UI.Controls.BocListImplementation.Rendering
+{
+  /// <summary>
+  /// <see cref="BocCellAttributeRenderingContext{TBocColumnDefinition}"/> wraps <see cref="BocColumnRenderingContext{TBocColumnDefinition}"/> for
+  /// use in <see cref="BocColumnRendererBase{TBocColumnDefinition}"/>.<see cref="BocColumnRendererBase{TBocColumnDefinition}.AddAttributesToRenderForTitleCell"/>
+  /// and in <see cref="BocColumnRendererBase{TBocColumnDefinition}"/>.<see cref="BocColumnRendererBase{TBocColumnDefinition}.AddAttributesToRenderForDataCell"/>.
+  /// </summary>
+  public readonly struct BocCellAttributeRenderingContext<TBocColumnDefinition>
+      where TBocColumnDefinition : BocColumnDefinition
+  {
+    private readonly BocColumnRenderingContext<TBocColumnDefinition> _renderingContext;
+
+    public BocCellAttributeRenderingContext (BocColumnRenderingContext<TBocColumnDefinition> renderingContext)
+    {
+      ArgumentUtility.CheckNotNull("renderingContext", renderingContext);
+
+      _renderingContext = renderingContext;
+    }
+
+    public HttpContextBase HttpContext => _renderingContext.HttpContext;
+
+    public IBocList Control => _renderingContext.Control;
+
+    public BusinessObjectWebServiceContext BusinessObjectWebServiceContext => _renderingContext.BusinessObjectWebServiceContext;
+
+    public int ColumnIndex => _renderingContext.ColumnIndex;
+
+    public int VisibleColumnIndex => _renderingContext.VisibleColumnIndex;
+
+    public TBocColumnDefinition ColumnDefinition => _renderingContext.ColumnDefinition;
+
+    public void AddAttributeToRender (string name, string value)
+    {
+      ArgumentUtility.CheckNotNull("name", name);
+
+      _renderingContext.Writer.AddAttribute(name, value);
+    }
+
+    public void AddAttributeToRender (string name, string value, bool fEncode)
+    {
+      ArgumentUtility.CheckNotNull("name", name);
+
+      _renderingContext.Writer.AddAttribute(name, value, fEncode);
+    }
+
+    public void AddAttributeToRender (string name, PlainTextString value)
+    {
+      ArgumentUtility.CheckNotNull("name", name);
+
+      value.AddAttributeTo(_renderingContext.Writer, name);
+    }
+
+    public void AddAttributeToRender (HtmlTextWriterAttribute name, string value)
+    {
+      ArgumentUtility.CheckNotNull("name", name);
+
```

Comment:
```suggestion
```

That's an enum (= value type). Wwe don't need the argument null check. please check the other overloads, too.